pf/tests/schema_test.go

```diff
@@ -59,3 +64,80 @@ func TestSchemaGen(t *testing.T) {
 		assert.Equal(t, "boolean", actionParameterPhases.Properties["p2"].Type)
 	})
 }
+
+func TestSchemaGenInSync(t *testing.T) {
+	type testCase struct {
+		name     string
+		file     string
+		pf       tfpf.ProviderInfo
+		provider tfbridge.ProviderInfo
+	}
+	testprovider.MuxedRandomProvider()
+
+	testCases := []testCase{
+		{
+			name: "tls",
+			file: "internal/testprovider/cmd/pulumi-resource-tls/schema.json",
+			pf:   testprovider.TLSProvider(),
+		},
+		{
+			name:     "muxedrandom",
+			file:     "internal/testprovider/cmd/pulumi-resource-muxedrandom/schema.json",
+			provider: testprovider.MuxedRandomProvider(),
+		},
+		{
+			name: "random",
+			file: "./internal/testprovider/cmd/pulumi-resource-random/schema.json",
+			pf:   testprovider.RandomProvider(),
+		},
+		{
+			name: "testbridge",
+			file: "./internal/testprovider/cmd/pulumi-resource-testbridge/schema.json",
+			pf:   testprovider.SyntheticTestBridgeProvider(),
+		},
+	}
+
+	renorm := func(a schema.PackageSpec) (out schema.PackageSpec) {
+		b, err := json.Marshal(a)
+		require.NoError(t, err)
+		err = json.Unmarshal(b, &out)
+		require.NoError(t, err)
+		return
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		expectedBytes, err := os.ReadFile(tc.file)
+		require.NoError(t, err)
+
+		var expectedSpec schema.PackageSpec
+		require.NoError(t, json.Unmarshal(expectedBytes, &expectedSpec))
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			var actualSpec schema.PackageSpec
+			if tc.pf.NewProvider != nil {
+				data := genMetadata(t, tc.pf)
+				require.NoError(t, json.Unmarshal(data.PackageSchema, &actualSpec))
+			} else {
+				var err error
+				actualSpec, err = tfgen.GenerateSchema(tc.provider, testSink(t))
+				require.NoError(t, err)
+			}
+
+			// Ignoring version differences, for some obscure reason they diverge right now.
```

Comment:
This should probably be an issue.

Reply:
Well I don't think it affects users in any way.